src/lib/components/collections/collections-list/ImportCollection.svelte

```diff
@@ -31,20 +31,67 @@
   let isSyntaxError: boolean = false;
   let isResponseLoading: boolean = false;
   let importData: string = "";
-  let importFile: File;
 
-  let fileName = "";
-  function handleFilesSelect(event: CustomEvent) {
-    fileName = event.detail ? event.detail.acceptedFiles[0]?.name : "";
-    importFile = event.detail ? event.detail.acceptedFiles[0] : "";
-  }
   const handleError = () => {
     isSyntaxError = false;
     isDataEmpty = false;
   };
   const handleInputField = (e) => {
     importData = e.target.value;
   };
+  let uploadCollection = {
+    file: {
+      value: [],
+      invalid: false,
+      showFileSizeError: false,
+      showFileTypeError: false,
+    },
+  };
+  const handleLogoInputChange = (
+    e: any,
+    maxSize: number,
+    supportedFileTypes: string[],
+  ) => {
+    if (
+      (e?.target?.files && e?.target?.files[0].size > maxSize * 1024) ||
+      (e?.dataTransfer?.files &&

```

Comment:
if you are using this at multiple places (line 58 also ) extract it to a const and then use it here like const filesize  or extract const filesize function to reuse.

Reply:
resolved